Libplanet.Extensions.Cocona/Commands/ApvCommand.cs

```diff
@@ -263,7 +267,32 @@ void TreeIntoTable(IValue tree, List<(string, string)> table, string key)
                 TreeIntoTable(v.Extra, data, "extra");
             }
 
-            Utils.PrintTable(("Field", "Value"), data);
+            if (json)
+            {
+                Console.WriteLine(JsonSerializer.Serialize(data.ToDictionary(e => e.Item1, e => e.Item2)));
+            }
+            else
+            {
+                Utils.PrintTable(("Field", "Value"), data);
+            }
+        }
+
+        [Command(Description = "Query app protocol version (a.k.a. APV) of target node.")]
+        public void Query(
+            [Argument(
+                Name = "TARGET",
+                Description = "Comma seperated peer information of target node.")]
+            string peerInfo)
+        {
+            try
+            {
+                BoundPeer peer = BoundPeer.ParsePeer(peerInfo);
+                Console.WriteLine(peer.QueryAppProtocolVersion().Token);
+            }
+            catch (Exception e)
+            {
+                throw Utils.Error($"Unexpected error occurred. [{e}]");
```

Comment:
For few expected exceptions, it could be explicitly caught so that more user-friendly error message is printed instead of internal exception trace. E.g. “invalid format of peer.”

Reply:
Applied d1e909cf8bb2e82770d03cb3e3ac24bb86ce5d89

Reply:
Applied https://github.com/planetarium/libplanet/pull/1240/commits/76df15f3aba8ec3a586f6ffaef58c022f1bc6eda